src/shims/unix/mem.rs

```diff
@@ -0,0 +1,214 @@
+use crate::machine::Mapping;
+use crate::shims::unix::fs::EvalContextExt as _;
+use crate::*;
+use rustc_target::abi::{Align, Size};
+
+const PAGE_SIZE: u64 = 4096;
+
+impl<'mir, 'tcx: 'mir> EvalContextExt<'mir, 'tcx> for crate::MiriEvalContext<'mir, 'tcx> {}
+pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx> {
+    fn mmap(
+        &mut self,
+        addr: &OpTy<'tcx, Provenance>,
+        length: &OpTy<'tcx, Provenance>,
+        prot: &OpTy<'tcx, Provenance>,
+        flags: &OpTy<'tcx, Provenance>,
+        fd: &OpTy<'tcx, Provenance>,
+        offset: &OpTy<'tcx, Provenance>,
+    ) -> InterpResult<'tcx, Pointer<Option<Provenance>>> {
+        let this = self.eval_context_mut();
+
+        let addr = this.read_pointer(addr)?;
+        let length = this.read_scalar(length)?.to_machine_usize(this)?;
+        let prot = this.read_scalar(prot)?.to_i32()?;
+        let flags = this.read_scalar(flags)?.to_i32()?;
+        let fd = this.read_scalar(fd)?.to_i32()?;
+        let offset = this.read_scalar(offset)?.to_machine_usize(this)?;
+
+        let print_args = || {
+            eprintln!(
+                "mmap(addr: {addr}, length: {length}, prot: {prot:x}, flags: {flags:0x}, fd: {fd}, offset: {offset})"
+            );
+        };
```

Comment:
I'm assuming this is a debugging helper and not meant to actually be shown to the user once this PR is ready?